src/redux/books/bookSlice.js

```diff
@@ -1,50 +1,122 @@
+
+    const myTransData = [];
+
+    Object.keys(myData).forEach((key) => {
+      const myTmpObj = myData[key][0];
+      myTmpObj.id = key;
+      myTransData.push(myTmpObj);
+    });
+
+    return myTransData;
+  } catch (error) {
+    return thunkAPI.rejectWithValue('something went wrong...');
+  }
+});
+
+export const postNewBook = createAsyncThunk('result/postNewBook', async ({ myTitle, myAuthor, myCategory }, thunkAPI) => {
+  const postBooksCreateUrl = `${url}${booksEndPoint}`;
+  const payload = {
+    item_id: uuidv4(),
+    title: myTitle,
+    author: myAuthor,
+    category: myCategory,
+  };
+  try {
+    const resp = await axios.post(postBooksCreateUrl, payload);
+    if (resp.data === 'Created') {
+      return payload;
+    }
+    return thunkAPI.rejectWithValue('something went wrong...');
+  } catch (error) {
+    return thunkAPI.rejectWithValue('something went wrong...');
+  }
+});
+
+export const deleteBook = createAsyncThunk('result/deleteBook', async ({ id }, thunkAPI) => {
+  const postBooksDeleteUrl = `${url}${booksEndPoint}/${id}`;
+  const payload = {
+    item_id: id,
+  };
+  console.log(payload);
+  try {
+    const resp = await axios.delete(postBooksDeleteUrl, payload);
+    console.log(resp);
+    if (resp.data === 'The book was deleted successfully!') {
+      return id;
+    }
+    return thunkAPI.rejectWithValue('something went wrong...');
+  } catch (error) {
+    return thunkAPI.rejectWithValue('something went wrong...');
+  }
+});
+
 const bookSlice = createSlice({
   name: 'book',
   initialState,
   reducers: {
     addBook: (state, action) => {
       console.log('adding');
-      state.bookItems = [
-        ...state.bookItems,
-        {
-          item_id: uuidv4(),
-          title: action.payload.title,
-          author: action.payload.author,
-          category: action.payload.category,
-        },
-      ];
+      state.books.push(action.payload);
     },
     removeBook: (state, action) => {
       const bookId = action.payload.itemId;
       state.bookItems = state.bookItems.filter((book) => book.item_id !== bookId);
     },
   },
+  extraReducers: {
+    [getResultItems.pending]: (state) => {
+      state.isLoading = true;
+    },
+    [getResultItems.fulfilled]: (state, action) => {
+      state.isLoading = false;
+      state.books = action.payload;
+    },
+    [getResultItems.rejected]: (state) => {
+      state.isLoading = false;
+    },
+    [postNewBook.pending]: (state) => {
+      state.isLoading = true;
+    },
+    [postNewBook.fulfilled]: (state, action) => {
+      state.isLoading = false;
+      state.books = [
+        ...state.books,
+        action.payload,
+      ];
+    },
+    [postNewBook.rejected]: (state) => {
+      state.isLoading = false;
+    },
+    [deleteBook.pending]: (state) => {
+      state.isLoading = true;
+    },
+    [deleteBook.fulfilled]: (state, action) => {
+      state.isLoading = false;
+      console.log(`my payload: ${action.payload}`);
```

Comment:
- [ ] Kindly remove this console.log and the others from your project as it is unprofessional to have this show up when a project has been deployed to production 👍